core-root/navigation/EditEvent.tsx

```diff
@@ -0,0 +1,110 @@
+  WithAlphaRegistrationProps
+} from "@core-root/AlphaRegister"
+import { EditEventFormDismissButton } from "@edit-event-boundary/Dismiss"
+import { EditEventView } from "@edit-event-boundary/EditEvent"
+import { editEventFormValueAtoms } from "@edit-event-boundary/FormAtoms"
+import {
+  RouteableEditEventFormValues,
+  fromRouteableEditFormValues
+} from "@event/EditFormValues"
+import {
+  LocationsSearchView,
+  useLocationsSearch
+} from "@location-search-boundary"
+import { StaticScreenProps, useNavigation } from "@react-navigation/native"
+import { createNativeStackNavigator } from "@react-navigation/native-stack"
+import { EventID } from "TiFShared/domain-models/Event"
+import { useSetAtom } from "jotai"
+import { StyleSheet } from "react-native"
+import { eventDetailsScreens } from "./EventDetails"
+import { profileScreens } from "./Profile"
+
+type EditEventScreenProps = WithAlphaRegistrationProps<
+  StaticScreenProps<RouteableEditEventFormValues & { id?: EventID }>
+>
+
+const EditEventScreen = withAlphaRegistration(
+  ({ session, route }: EditEventScreenProps) => {
+    const navigation = useNavigation()
+    const { pushEventDetails } = useCoreNavigation()
+    return (
+      <EditEventView
+        eventId={route.params.id}
+        initialValues={fromRouteableEditFormValues(route.params)}
+        hostName={session.name}
+        hostProfileImageURL={session.profileImageURL}
+        onSelectLocationTapped={() => {
+          navigation.navigate("editEvent", {
+            screen: "editEventLocationSearch"
+          })
+        }}
+        onSuccess={(e) => pushEventDetails(e.id, "replace")}
+        style={styles.screen}
+      />
+    )
+  }
+)
+
+const LocationSearchScreen = () => {
+  const setLocation = useSetAtom(editEventFormValueAtoms.location)
+  const navigation = useNavigation()
+  return (
+    <LocationsSearchView
+      state={useLocationsSearch()}
+      onUserLocationSelected={(location) => {
+        setLocation({ coordinate: location.coords, placemark: undefined })
+        navigation.goBack()
+      }}
+      onLocationSelected={(location) => {
+        setLocation(location)
+        navigation.goBack()
+      }}
+    />
+  )
+}
+
+const EditEventFormBackButton = () => (
+  <EditEventFormDismissButton onDismiss={useNavigation().goBack} />
+)
+
+export const EditEventNavigator = createNativeStackNavigator({
+  screenOptions: () => BASE_HEADER_SCREEN_OPTIONS,
+  screens: {
+    editEventForm: {
+      options: {
+        headerTitle: "Edit Event",
+        headerLeft: EditEventFormBackButton
+      },
+      screen: EditEventScreen
+    },
+    createEventForm: {
+      options: {
+        headerTitle: "Create Event",
+        headerLeft: EditEventFormBackButton
+      },
+      screen: EditEventScreen
+    },
+    editEventLocationSearch: {
+      options: { headerShown: false },
+      screen: LocationSearchScreen
+    },
+    ...eventDetailsScreens(),
+    ...profileScreens()
+  }
+})
+
```

Comment:
So wait, editEvent and createEvent would use the same screen? Or for, createEvent, is this just a placeholder?

Reply:
They use the same screen, but with different screen options (notably the title being different). We could detect event creation vs editing in `editEventForm` and update the header dynamically, but this is annoying to do and it's easier to use 2 routes if there are only 2 static screen option configurations.